packages/cozy-sharing/src/components/ShareByEmail.jsx

```diff
@@ -201,81 +190,83 @@ class ShareByEmail extends Component {
-      const cozyUrl = Contact.getPrimaryCozy(recipient)
-
-      if (email) {
-        return [
-          `${documentType}.share.shareByEmail.success`,
-          {
-            email
-          }
-        ]
-      } else if (cozyUrl) {
-        return [
-          `${documentType}.share.shareByEmail.success`,
-          {
-            email: cozyUrl
-          }
-        ]
-      } else {
-        return [
-          `${documentType}.share.shareByEmail.genericSuccess`,
-          {
-            count: 1
-          }
-        ]
-      }
-    } else {
-      return [
-        `${documentType}.share.shareByEmail.genericSuccess`,
-        {
-          count: countNewRecipients(recipientsBefore, recipients)
-        }
-      ]
-    }
-  }
-
-  share = () => {
-    const { document, sharingDesc, onShare, createContact } = this.props
+  share = async () => {
+    const {
+      document,
+      sharingDesc,
+      onShare,
+      createContact,
+      documentType
+    } = this.props
     const { recipients, sharingType } = this.state
     if (recipients.length === 0) {
       return
     }
+    const client = useClient()
 
     // we can't use currentRecipients prop in getSuccessMessage because it may use
     // the updated prop to count the new recipients
     const recipientsBefore = this.props.currentRecipients
 
+    const verifiedContacts = []
     this.setState(state => ({ ...state, loading: true }))
-    Promise.all(
-      recipients.map(recipient =>
-        recipient.id
-          ? recipient
-          : createContact({
-              email: [{ address: recipient.email, primary: true }]
-            }).then(resp => resp.data)
-      )
+
+    await Promise.all(
+      recipients.map(async recipient => {
+        if (recipient.id) {
+          verifiedContacts.push(recipient)
+        } else if (validateEmail(recipient.email)) {
+          const contact = await client.collection('io.cozy.contacts').find(
+            {
+              email: {
+                $elemMatch: {
+                  address: recipient.email
+                }
+              },
+              id: {
+                $gt: null
+              }
+            },
+            {
+              indexedFields: ['id']
+            }
+          )
+          if (contact.data.length > 0) {
+            //We take the shortcut that if we have sevaral contacts
+            //with the same address, we take the first one for now
+            verifiedContacts.push(contact.data[0])
+          } else {
+            verifiedContacts.push(recipient)
+          }
+        }
+      })
     )
-      .then(recipients =>
-        onShare(document, recipients, sharingType, sharingDesc)
+    try {
+      const allCreatedRecipients = await Promise.all(
+        verifiedContacts.map(recipient =>
+          recipient.id
+            ? recipient
+            : createContact({
+                email: [{ address: recipient.email, primary: true }]
+              }).then(resp => resp.data)
```

Comment:
Could we use async/await here?

Reply:
old commit. already fixed in a more recent commit ;)